dashboard/src/api/client/query.tsx

```diff
@@ -0,0 +1,57 @@
+import { QueryCache, QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import { AxiosError } from 'axios';
+import { StatusCodes } from 'http-status-codes';
+import React from 'react';
+import { toast } from 'react-toastify';
+
+export interface GeneralAPIResponse {
+    error?: {
+        code: number;
+        message: string;
+        errors: { reason: string; message: string }[];
+    };
+}
+
+// Create react-query client
+const queryClient = new QueryClient({
+    defaultOptions: {
+        queries: {
+            staleTime: 1000 * 60 * 15, // 15 minutes
+            gcTime: 1000 * 60 * 15, // 15 minutes
+            notifyOnChangeProps: ['data', 'isLoading', 'error'],
+            retry: (retryCount, err) => {
+                // Get status from error object
+                let status: number | undefined = undefined;
+
+                // General API error setup
+                if (Object.prototype.hasOwnProperty.call(err, 'code')) {
+                    status = (err as GeneralAPIResponse).error?.code;
+                }
+
+                // Normal Axios error setup
+                if (Object.prototype.hasOwnProperty.call(err, 'response')) {
+                    status = (err as AxiosError).response?.status;
+                }
+
+                // Do not retry if status is Unauthorized or Forbidden
+                if (status !== undefined && [StatusCodes.UNAUTHORIZED, StatusCodes.FORBIDDEN].includes(status)) {
+                    return false;
+                }
+
+                // Default 3 tries (initial + 2 retries)
+                return retryCount < 2;
+            },
+        },
+    },
+    queryCache: new QueryCache({
+        onError: (error) => toast.error(`Something went wrong: ${error.message}`),
+    }),
+});
+
+
+// TODO: definitely not the way to o it(children) but will figure out later
+const ReactQueryClientProvider: React.FC<{ children: React.ReactNode }> = ({ children }) => {
```

Comment:
`React.FC` has children included by default
```suggestion
const ReactQueryClientProvider: React.FC = ({ children }) => {
```

Reply:
tried initially but getting type error: Property 'children' does not exist on type '{}'.

Reply:
That's weird considering the type below.
![image](https://github.com/UNIwise/parrot/assets/22818957/bf0fa767-f85e-4d0c-9104-e6d68f7d6050)